File: 1768. Merge Strings Alternately/GenSol.ts
const mergeAlternately = (word1: string, word2: string): string => {
    let res: string = ""
    let i: number = 0

    while (i < word1.length || i < word2.length) {
        if (i < word1.length) res += word1[i]
        if (i < word2.length) res += word2[i]
        i += 1
    }

    return res
}

File: 1768. Merge Strings Alternately/GenSol.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        '''
        String processing & array/list

        Handle unmatched string length
        '''
        i = 0
        res = ""
        while i < len(word1) and i < len(word2):
            res += word1[i]
            res += word2[i]
            i += 1
        if i == len(word1):
            res += word2[i:]
        if i == len(word2):
            res += word1[i:]
        return res